benchmarks/petsc-bps.sh

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+# All Rights reserved. See files LICENSE and NOTICE for details.
+#
+# This file is part of CEED, a collection of benchmarks, miniapps, software
+# libraries and APIs for efficient high-order finite element and spectral
+# element discretizations for exascale applications. For more information and
+# source code availability see http://github.com/ceed.
+#
+# The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+# a collaborative effort of two U.S. Department of Energy organizations (Office
+# of Science and the National Nuclear Security Administration) responsible for
+# the planning and preparation of a capable exascale ecosystem, including
+# software, applications, hardware, advanced system engineering and early
+# testbed platforms, in support of the nation's exascale computing imperative.
+
+
+function run_tests()
+{
+   $dry_run cd "$test_exe_dir"
+
+   # Some of the available options are:
+   # -degree <1>: Polynomial degree of tensor product basis
+   # -qextra <2>: Number of extra quadrature points
+   # -ceed </cpu/self>: CEED resource specifier
+   # -cells <10,10,10>: Target number of locally (per rank) owned mesh cells
+
+   # The variables 'max_dofs_node', and 'max_p' can be set on the command line
+   # invoking the 'benchmark.sh' script.
+   local ceed="${ceed:-/cpu/self}"
+   local common_args=(-ceed $ceed -pc_type none -benchmark)
+   local max_dofs_node_def=$((3*2**20))
+   local max_dofs_node=${max_dofs_node:-$max_dofs_node_def}
+   local max_loc_dofs=$((max_dofs_node/num_proc_node))
+   local max_p=${max_p:-8}
+   local sol_p=
+   for ((sol_p = 1; sol_p <= max_p; sol_p++)); do
+      local loc_el=
+      for ((loc_el = 1; loc_el*sol_p**3 <= max_loc_dofs; loc_el = 2*loc_el)); do
+         local loc_dofs=$((loc_el*sol_p**3))
+         local loc_cell=$(echo "e(l($loc_dofs)/3)" | bc -l)
+         loc_cell=${loc_cell%.*}
+         local all_args=("${common_args[@]}" -degree $sol_p -cells $loc_cell,$loc_cell,$loc_cell -problem $bp)
```

Comment:
`-cells` is global, but this looks like it's used locally. Perhaps just do this processing in the C executable? E.g., accept `-dofs_local` and compute a problem size using `MPI_Comm_size(comm, &size)` and `total_cells = dofs_local * size / (ncomp * CeedPowInt(degree, 3))`, then compute suitable box sizes?

Reply:
I have updated it using the same strategy that is in `bpsraw.c`